### PR TITLE
[mds-*] Cleanup eslint-disable usage

### PR DESCRIPTION
**Pull request by avatarneil**
Re-enable eslint directives where possible by refactoring code, and add eslint-reason for any outstanding disables.

## PR Checklist

 - [x] simple searchable title - `[mds-db] Add PG env var`, `[config] Fix eslint config`
 - [x] briefly describe the changes in this PR
 - [ ] mark as draft if should not be merged
 - [ ] write tests for all new functionality

## Impacts
- [x] Provider
- [x] Agency
- [x] Audit
- [x] Policy
- [x] Compliance

### CODE REVIEW

**Comment by avatarneil** [comment]
Ser Jenkins, please run tests <3

File: packages/mds-db/sql-utils.ts [hunk 1]
```diff
@@ -138,18 +133,18 @@ export function logSql(sql: string, ...values: unknown[]): void {
   if (!pgDebug) {
     return
   }
-  /* eslint-disable no-param-reassign */
+  let out: unknown
```
Comment (mdurling) [inline packages/mds-db/sql-utils.ts]: Looks like it's at least `unknown[]`

**Comment by avatarneil** [comment]
Mr. Jenkins... Please run tests, and let me know that this code is the best!

**Comment by avatarneil** [comment]
Mr. Jenkins... Please run tests, and let me know that this code is the best!

**Pull request merged**